python_modules/dagster/dagster/core/log.py

```diff
@@ -80,14 +83,25 @@ def _log(self, method, orig_message, message_props):
         #     message_with_structured_props, extra={DAGSTER_META_KEY: all_props}
         # )
 
+        logger_found = False
         for logger in self.loggers:
-            logger_method = check.is_callable(getattr(logger, method))
-            if logger.name == DAGSTER_DEFAULT_LOGGER:
-                logger_method(
-                    msg_with_multiline_structured_props, extra={DAGSTER_META_KEY: all_props}
-                )
-            else:
-                logger_method(msg_with_structured_props, extra={DAGSTER_META_KEY: all_props})
+            if hasattr(logger, method):
+                logger_found = True
+                logger_method = check.is_callable(getattr(logger, method))
+                if logger.name == DAGSTER_DEFAULT_LOGGER:
+                    logger_method(
+                        msg_with_multiline_structured_props, extra={DAGSTER_META_KEY: all_props}
+                    )
+                else:
+                    logger_method(msg_with_structured_props, extra={DAGSTER_META_KEY: all_props})
+
+        if not logger_found:
```

Comment:
so the issue for this method is that will be possible to induce crashes at runtime by changing logging config, which I am pretty concerned about.

Reply:
i'm not 100% certain which the cases you're thinking about are --

047d55 reinstates and documents the status quo ante behavior when there are *no* loggers configured (silently passes)

if i have custom loggers which define `.foo()` configured, and call `context.log.foo()` somewhere in my codebase, bnut then remove the custom loggers, the behavior in this diff is that we log an `ERROR` (and continue to execute) -- we could log this as a `WARNING` instead; previous behavior if i called `context.log.foo` from user code was:

```       dagster_event = {"event_specific_data": [["AttributeError: 'DagsterLog' object has no attribute 'gargle'\n", ["  File \"/Users/max/dev/dagster/python_modules/dagster/dagster/core/errors.py\", line 177, in user_code_error_boundary\n    yield\n", "  File \"/Users/max/dev/dagster/python_modules/dagster/dagster/core/execution_plan/simple_engine.py\", line 374, in _iterate_step_outputs_within_boundary\n    for step_output in gen:\n", "  File \"/Users/max/dev/dagster/python_modules/dagster/dagster/core/execution_plan/transform.py\", line 88, in _execute_core_transform\n    for step_output in _yield_transform_results(transform_context, inputs):\n", "  File \"/Users/max/dev/dagster/python_modules/dagster/dagster/core/execution_plan/transform.py\", line 52, in _yield_transform_results\n    for result in gen:\n", "  File \"/Users/max/dev/dagster/python_modules/dagster/dagster/core/definitions/decorators.py\", line 310, in transform\n    result = fn(context, **kwargs)\n", "  File \"/Users/max/dev/dagster/python_modules/dagster/dagster_tests/core_tests/test_pipeline_errors.py\", line 283, in bad_logger\n    context.log.gargle('bargle')\n"], "AttributeError"]], "event_type_value": "STEP_FAILURE", "pipeline_name": "test_bad_log", "solid_definition_name": "bad_logger", "solid_name": "bad_logger", "step_key": "bad_logger.transform", "step_kind_value": "TRANSFORM", "tags": {"pipeline": "test_bad_log", "solid": "bad_logger", "solid_definition": "bad_logger", "step_key": "bad_logger.transform"}}```

Reply:
Right I guess what I'm saying is that I'm a bit concerned about the ability to inject errors into the pipeline for what looks like a totally valid configuration change. My instinct is that if you are provided a set of mixed loggers we should force the user to either 1) define some interface that maps a well known set of methods to the correct calls in the underlying loggers to 2) expose some explicit API that does the moral equivalent of "try to log to this specific logger but only if it exists"